samples/java-protobuf-tracing/src/main/java/com/example/FactorialControllerAction.java

```diff
@@ -0,0 +1,103 @@
+package com.example;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import kalix.javasdk.action.ActionCreationContext;
+
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import com.example.FactorialActionApi.MessageResponse;
+import com.google.protobuf.Empty;
+
+
+// This class was initially generated based on the .proto definition by Kalix tooling.
+// This is the implementation for the Action Service described in your com/example/factorial_action.proto file.
+//
+// As long as this file exists it will not be overwritten: you can maintain it yourself,
+// or delete it so it is regenerated as needed.
+
+public class FactorialControllerAction extends AbstractFactorialControllerAction {
```

Comment:
I don't understand the idea of naming this FactorialController, we just call some random service here.

Reply:
That's outdated. Now is called `ControllerAction`.